applications/asset_tracker/src/ui/include/ui.h

```diff
@@ -0,0 +1,226 @@
+	UI_CLOUD_CONNECTED	= UI_LED_ON(DK_LED4_MSK),
+	UI_CLOUD_PAIRING	= UI_LED_BLINK(DK_LED3_MSK | DK_LED4_MSK),
+	UI_ACCEL_CALIBRATING	= UI_LED_ON(DK_LED1_MSK | DK_LED3_MSK),
+	UI_LED_ERROR_CLOUD	= UI_LED_BLINK(DK_LED1_MSK | DK_LED4_MSK),
+	UI_LED_ERROR_BSD_REC	= UI_LED_BLINK(DK_LED1_MSK | DK_LED3_MSK),
+	UI_LED_ERROR_BSD_IRREC	= UI_LED_BLINK(DK_ALL_LEDS_MSK),
+	UI_LED_ERROR_LTE_LC	= UI_LED_BLINK(DK_LED1_MSK | DK_LED2_MSK),
+	UI_LED_ERROR_UNKNOWN	= UI_LED_ON(DK_ALL_LEDS_MSK)
+#endif /* CONFIG_UI_LED_USE_PWM */
+};
+
+/**@brief UI event types. */
+enum ui_evt_type {
+	UI_EVT_BUTTON_ACTIVE,
+	UI_EVT_BUTTON_INACTIVE
+};
+
+/**@brief UI event structure. */
+struct ui_evt {
+	enum ui_evt_type type;
+
+	union {
+		u32_t button;
+	};
+};
+
+/**
+ * @brief UI callback handler type definition.
+ *
+ * @param evt Pointer to event struct.
+ */
+typedef void (*ui_callback_t)(struct ui_evt evt);
+
+/**
+ * @brief Initializes the user interface module.
+ *
+ * @param cb UI callback handler. Can be NULL to disable callbacks.
+ *
+ * @return 0 on success or negative error value on failure.
+ */
+int ui_init(ui_callback_t cb);
+
+/**
+ * @brief Sets the LED pattern.
+ *
+ * @param pattern LED pattern.
+ */
+void ui_led_set_pattern(enum ui_led_pattern pattern);
+
+/**
+ * @brief Sets a LED's state. Only the one LED is affected, the rest of the
+ *	  LED pattern is preserved. Only has effect if the specified LED is not
+ *	  controlled by PWM.
+ *
+ * @param led LED number to be controlled.
+ * @param value 0 turns the LED off, a non-zero value turns the LED on.
+ */
+void ui_led_set_state(u32_t led, u8_t value);
+
+/**
+ * @brief Gets the LED pattern.
+ *
+ * @return Current LED pattern.
+ */
+enum ui_led_pattern ui_led_get_pattern(void);
+
+/**
+ * @brief Sets the LED RGB color.
+ *
+ * @param red Red, in range 0 - 255.
+ * @param green Green, in range 0 - 255.
+ * @param blue Blue, in range 0 - 255.
+ *
+ * @return 0 on success or negative error value on failure.
+ */
+int ui_led_set_color(u8_t red, u8_t green, u8_t blue);
+
+/**
+ * @brief Get the state of a button.
+ *
+ * @param button Button number.
+ *
+ * @return 1 if button is active, 0 if it's inactive.
+ */
+bool ui_button_is_active(u32_t button);
+
+/**
+ * @brief Set the buzzer frequency.
+ *
+ * @param frequency Frequency. If set to 0, the buzzer is disabled.
+ * 		    The frequency is limited to the range 100 - 10 000 Hz.
+ * @param intensity Intensity of the buzzer output. If set to 0, the buzzer is
+ * 		    disabled.
+ * 		    The frequency is limited to the range 0 - 100 %.
+ *
+ * @return 0 on success or negative error value on failure.
+ */
+int ui_buzzer_set_frequency(u32_t frequency, u8_t intensity);
+
+/**
+ * @brief Write value to pin controlling NMOS transistor.
+ *
+ * @param nmos_idx	NMOS to control.
+ * @param value		1 sets high signal on NMOS gate, 0 sets it low.
+ *
+ * @return 0 on success or negative error value on failure.
+ */
+int ui_nmos_write(size_t nmos_idx, u8_t value);
```

Comment:
Possibly. The `nmos` name is indeed PCA20035 specific. I'll put on my thinking hat and see if I can come up with something else, otherwise I'm fine with `ui_pin_` prefix.